Wrapping/Generators/Python/itk/support/itkExtras.py

```diff
@@ -773,6 +773,58 @@ def meshread(
     reader.Update()
     return reader.GetOutput()
 
+def transformread(filename: fileiotype):
+    """Read an itk Transform file.
+
+    Parameters
+    ----------
+
+    filename:
+        Path to the transform file (typically a .h5 file).
+
+    Returns
+    -------
+
+    A Python list containing the transforms in the file.
+    """
+    import itk
+
+    reader = itk.TransformFileReaderTemplate[itk.D].New()
+    reader.SetFileName(str(filename))
+    reader.Update()
+
+    transforms = []
+    transform_list = reader.GetModifiableTransformList()
+    while not transform_list.empty():
+        transform = transform_list.pop()
+        transforms.append(itk.down_cast(transform))
```

Comment:
My Python knowledge is limited, but I just wonder, would it be possible to retrieve the transforms without modifying (i.e. without popping) the `transform_list`?  Something like the following:

    for transform in transform_list:
        transforms.append(itk.down_cast(transform))

Reply:
The transform_list is a C++-wrapped `std::list` and the `transforms` are a Python `list`. This syntax was required to procure the elements with SWIG's wrapping of the `std::list`, which leaves a little to be desired. It would be nice to not modify, but we discard the `transform_list` and `reader` at the end of the function, so it is not necessary.

Reply:
Basically I would like to be sure that the order is preserved correctly when you do `transform_list.pop` and `transforms.append`?

Reply:
Good point!

Checking the [docs](https://docs.python.org/3/tutorial/datastructures.html), it is at the end, so we have to reverse the result.

Reply:
@thewtex I'm glad I could be of help. But if `transform_list` is just a regular Python list, why not use the simple `for transform in transform_list`, as I suggested? 

    for transform in transform_list:
        transforms.append(itk.down_cast(transform))

?

Reply:
This is a good idea, and I tried it at first, but the SWIG wrapping is not functioning correctly to support it.  `transform_list` is a SWIG wrapped `std::list` that also tries to expose the Python [Abstract Base Clase (ABC)](https://docs.python.org/3/library/collections.abc.html) interface for a MutableSequence. However, when attempting to iterate over it in this way, it does not return the correctly typed `itk.TransformBaseTemplate` object that we can use.

Reply:
Thanks, Matt. Do you possibly also have a link to the API of the SWIG wrapped `std::list`?

Reply:
Here's how to produce what is generated, this time for a `double`:

```python
import itk
l = itk.list[itk.D]()
dir(l)
```

The result:

```
['__abstractmethods__', '__bool__', '__class__', '__contains__',
'__delattr__', '__delitem__', '__delslice__', '__dict__', '__dir__',
'__doc__', '__eq__', '__format__', '__ge__', '__getattribute__',
'__getitem__', '__getslice__', '__gt__', '__hash__', '__iadd__', '__init__',
'__init_subclass__', '__iter__', '__le__', '__len__', '__lt__', '__module__',
'__ne__', '__new__', '__nonzero__', '__reduce__', '__reduce_ex__', '__repr__',
'__reversed__', '__setattr__', '__setitem__', '__setslice__', '__sizeof__',
'__slots__', '__str__', '__subclasshook__', '__swig_destroy__', '__weakref__',
'_abc_impl', 'append', 'assign', 'back', 'begin', 'clear', 'count', 'empty',
'end', 'erase', 'extend', 'front', 'get_allocator', 'index', 'insert',
'iterator', 'pop', 'pop_back', 'pop_front', 'push_back', 'push_front',
'rbegin', 'remove', 'rend', 'resize', 'reverse', 'size', 'swap', 'this',
'thisown']

```

Reply:
Ah, I see. `pop` seems somewhat special, in this "dir listing", because it isn't part of the C++ `std::list` 🤔 It's a pity that regular Python `for` iteration over the list doesn't work properly. While I don't expect the C++ `begin` to `end` to work well either, for the SWIG wrapping.